### PR TITLE
Close #19003: Change _currentTrackCurve to a struct

### LINKED ISSUE #19003: Change _currentTrackCurve to a struct
Currently, bit 16 is set when a special track piece is selected, which complicates checks. Rather, it should be a struct with a track piece (16-bit unsigned int) and a bool. This struct may also be needed for some other variables, this needs to be checked.

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/scripting/bindings/ride/ScTrackSegment.cpp [hunk 1]
```diff
@@ -194,42 +194,46 @@ std::vector<DukValue> ScTrackSegment::getSubpositions(uint8_t trackSubposition,
     return result;
 }
 
+static DukValue _trackCurveToString(duk_context* ctx, TrackCurve curve)
```
Comment (AaronVanGeffen) [inline src/openrct2/scripting/bindings/ride/ScTrackSegment.cpp:197]: We have EnumMap for this purpose, right?

Comment (Gymnasiast) [reply src/openrct2/scripting/bindings/ride/ScTrackSegment.cpp:197]: That one works the other way.

Comment (AaronVanGeffen) [reply src/openrct2/scripting/bindings/ride/ScTrackSegment.cpp:197]: EnumMap is declared both ways. Note:
```cpp

    std::string_view operator[](T k) const
    {
        auto it = find(k);
        return it->first;
    }

    T operator[](std::string_view k) const
    {
        auto it = find(k);
        return it->second;
    }
```

**Review by AaronVanGeffen — APPROVED** [review]
Changes look fine, but I haven't tested them.

**Pull request merged**